lib/Conversion/TorchToTcp/Misc.cpp

```diff
@@ -90,10 +90,14 @@ class ConvertValueTensorLiteralOp
             ->convertType(op.getType())
             .cast<RankedTensorType>();
 
-    // TODO: Implement integer constant lowering to tcp::ConstOp
-    if (auto elements = op.getValueAttr().dyn_cast<DenseIntElementsAttr>())
-      return rewriter.notifyMatchFailure(
-          op, "Integer constants lowering to TCP is unimplemented");
+    if (auto elements = op.getValueAttr().dyn_cast<DenseIntElementsAttr>()) {
+      Type elementType = resultType.getElementType();
+      auto denseIntAttr = elements.mapValues(elementType, [&](const APInt &v) {
+        return APInt(elementType.getIntOrFloatBitWidth(), v.getSExtValue());
+      });
+      rewriter.replaceOpWithNewOp<tcp::ConstOp>(op, resultType, denseIntAttr);
```

Comment:
I'm assuming this handles both signed and unsigned integer cases? If so, could you add a lit test for `ui32` as well?

Reply:
Only `ui8` is the legal unsigned integer type via torch dialect.